src/webots/scene_tree/WbDoubleEditor.cpp

```diff
@@ -69,8 +69,9 @@ void WbDoubleEditor::resetFocus() {
 }
 
 void WbDoubleEditor::applyIfNeeded() {
-  if (field() && ((field()->hasRestrictedValues() && mDouble != mComboBox->currentText().toDouble()) ||
-                  (!field()->hasRestrictedValues() && mDouble != mSpinBox->value())))
+  if (field() &&
+      ((field()->hasRestrictedValues() && mDouble != mComboBox->currentText().toDouble()) ||
+       (!field()->hasRestrictedValues() && fabs(mDouble - mSpinBox->value()) > WbPrecision::epsilon(WbPrecision::DOUBLE_MAX))))
```

Comment:
```suggestion
       (!field()->hasRestrictedValues() && mDouble != mSpinBox->text().toDouble())))
```
Isn't it simpler and more consistent this way?